main.tf

```diff
@@ -70,20 +70,24 @@ resource "aws_cloudwatch_event_target" "this" {
   }
 
   dynamic "ecs_target" {
-    for_each = lookup(each.value, "ecs_target", null) != null ? [true] : []
+    for_each = lookup(each.value, "ecs_target", null) != null ? [
+      each.value.ecs_target
```

Comment:
`each.value.target` should be known for this `for_each` to work always. It is better to leave `true` as it was.

Reply:
the plan wouldn't succeed if I'd roll this back.

Reply:
Not sure why exactly but ok. :)